src/Proxy.hs

```diff
@@ -890,5 +893,47 @@ putGTUDropR addrText = do
 getGlobalFileR :: Handler TypedContent
 getGlobalFileR = toTypedContent . globalInfo <$> getYesod
 
+-- Queries the transaction database and the GRPC, 
+-- then if both succeed checks that the last final block is less than 5 minutes old.
+getHealthR :: Handler TypedContent
+getHealthR = 
+  runGRPC doGetBlockFinalInfo $ \case
+      Nothing -> do
+        $(logError) $ "Could not get response from GRPC."
+        sendResponse $ object $ ["healthy" .= False, "reason" .= ("Could not get response from GRPC.":: String)]
+      Just lastFinalBlockInfo -> do
+        $(logInfo) "Successfully got best block info."
+        result :: [(Entity Entry, Entity Summary)] <- runDB $ E.select $
+                  E.from $ \val -> do
+                  E.limit 1 -- query for any single row
+                  return val
+        case result of
+          [] ->
+            sendResponse $ object $ ["healthy" .= False, "reason" .= ("Could not get response from database.":: String)]
+          _ -> do
+            $(logInfo) "Successfully queried database."
+            -- get block slot time from block info object, compare with current time: reject if more than 5 minutes:
+            case lastFinalBlockInfo of
+              Object hm -> 
+                case HM.lookup "blockSlotTime" hm of
```

Comment:
In Concordium/Client/Cli.hs in concordium-client there is a type `BlockInfoResult` which you can use here to parse the response, instead of deconstructing the JSON manually.